sumologic/sumologic_sources.go

```diff
@@ -142,6 +143,12 @@ func resourceSumologicSource() *schema.Resource {
 					},
 				},
 			},
+			"hash_algorithm": {
+				Type: schema.TypeString,
+				Optional: true
+				Default: nil,
```

Comment:
should we change the default to "MD5" as per documentation?

Reply:
It's an optional field in source json. If you don't explicitly `hashAlgorithm`, it won't be serialized into source json. Service will do some internal work and  use `MD5` for its hash algorithms.

Some reason that I don't want to default to `MD5`.
1. This is added for all sources. However, it's only needed for hash rules, which won't apply to metric sources. I don't want this field to be shown for metrics sources.

2. For existing sources, if they don't mention `hashAlgorithm` in the source, it will still keep as it is and they won't notify for new changes needed to be applied.